### PR TITLE
[WIP] Expose group info to C API + other wrappers

### PR DESCRIPTION
**Pull request by Helw150**
Doing my best at taking over from @hcho3 @ #3514 

TODOs:

- [x] Add C++ test
- [ ] Expose the function to Python wrapper
- [ ] Expose the function to Java/Scala wrapper
- [x] Expose the function to R wrapper

### CODE REVIEW

**Comment by khotilov** [comment]
I've exposed it to R and have fixed the C++ test. Was the plan to use this same PR also for adding the Python/Java/Scala parts?

**Comment by hcho3** [comment]
@khotilov Thanks. I'll be adding Python part and @Helw150 Java/Scala

**Comment by Helw150** [comment]
I should be free up to work on the JNI and then Scala/Java stuff in the next couple of days! Thanks so much @khotilov for doing the heavy lifting

File: jvm-packages/xgboost4j/src/test/java/ml/dmlc/xgboost4j/java/DMatrixTest.java [hunk 1]
```diff
@@ -223,4 +223,29 @@ public void testCreateFromDenseMatrixWithMissingValue() throws XGBoostError {
     TestCase.assertTrue(dmat0.rowNum() == 10);
     TestCase.assertTrue(dmat0.getLabel().length == 10);
   }
+
+  @Test
+  public void testSetAndGetGroup() throws XGBoostError {
```
Comment (CodingCat) [inline jvm-packages/xgboost4j/src/test/java/ml/dmlc/xgboost4j/java/DMatrixTest.java]: would you add the corresponding test in scala pkg?

File: jvm-packages/xgboost4j/src/test/scala/ml/dmlc/xgboost4j/scala/DMatrixSuite.scala [hunk 2]
```diff
@@ -167,4 +167,52 @@ class DMatrixSuite extends FunSuite {
     assert(dmat0.rowNum === 10)
     assert(dmat0.getLabel.length === 10)
   }
+
+    test("create DMatrix from DenseMatrix") {
```
Comment (CodingCat) [inline jvm-packages/xgboost4j/src/test/scala/ml/dmlc/xgboost4j/scala/DMatrixSuite.scala]: indention here is incorrect,

Comment (hcho3) [reply jvm-packages/xgboost4j/src/test/scala/ml/dmlc/xgboost4j/scala/DMatrixSuite.scala]: @CodingCat Can we set up lint / style check for Java and Scala code? Currently, Python and C++ code are automatically checked for style.

Comment (Helw150) [reply jvm-packages/xgboost4j/src/test/scala/ml/dmlc/xgboost4j/scala/DMatrixSuite.scala]: @hcho3 There is linting that this PR is failing. The main thing is that you can pass the lint filter but not meet the style guide, e.g. my other PR where I used Scalafmt.

**Comment by stpyang** [comment]
I believe the java/scala group getters are now in this PR https://github.com/dmlc/xgboost/pull/4838

**Pull request closed without merging**